### ISSUE TITLE
rotation_matrix_to_quaternion never returns a unit quaternion at the default eps: R2q(I) = [1.0000000012499999, 0, 0, 0] because eps is added inside the sqrt

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#3951]
`rotation_matrix_to_quaternion` adds its `eps` **inside** the `sqrt`, so with the default `eps = 1e-8` it never returns a unit quaternion — not even for the identity matrix:

```python
>>> rotation_matrix_to_quaternion(torch.eye(3, dtype=torch.float64)).tolist()
[1.0000000012499999, 0.0, 0.0, 0.0]
>>> rotation_matrix_to_quaternion(torch.eye(3, dtype=torch.float64), eps=0.0).tolist()
[1.0, 0.0, 0.0, 0.0]
```

All four branches do it (`kornia/geometry/conversions.py`, `trace_positive_cond` and `cond_1..3`):

```python
sq = torch.sqrt(trace + 1.0 + eps) * 2.0        # sq = 4 * qw
...
sq = torch.sqrt(1.0 + m11 - m00 - m22 + eps) * 2.0  # sq = 4 * qy
```

Every component is scaled by `sqrt(1 + eps/s)`, i.e. inflated by `~eps/(2s)`, so `||q|| − 1 ≈ 1.25e-09` at the identity and up to `2.5e-09` over random exact rotations. The docstring describes the parameter as "small value to avoid zero division", but the divisions in this function are already protected — they go through `safe_zero_division`, which uses `torch.finfo(...).tiny`. This `eps` protects nothing; it only biases the value.

## Repro

```python
import torch
from kornia.geometry.conversions import (
    quaternion_to_rotation_matrix as q2R,
    rotation_matrix_to_quaternion as R2q,
)

I3 = torch.eye(3, dtype=torch.float64)
print("R2q(I)          =", R2q(I3).tolist())
print("R2q(I, eps=0.0) =", R2q(I3, eps=0.0).tolist())
print("||R2q(I)|| - 1  =", (R2q(I3).norm() - 1).item())

# statistics over 200 *exact* rotation matrices built by q2R of random unit quaternions
g = torch.Generator().manual_seed(5)
worst_rt, worst_norm = 0.0, 0.0
for _ in range(200):
    qq = torch.randn(4, generator=g, dtype=torch.float64)
    qq = qq / qq.norm()
    Rex = q2R(qq)                                   # exact: det=1, RR^T=I to 2e-16
    worst_rt = max(worst_rt, (q2R(R2q(Rex)) - Rex).abs().max().item())
    worst_norm = max(worst_norm, abs(R2q(Rex).norm().item() - 1.0))
print("worst |q2R(R2q(R)) - R| over 200 exact rotations :", worst_rt)
print("worst | ||R2q(R)|| - 1 | over the same 200       :", worst_norm)
```

The construction of the test matrices is load-bearing: they are built by `q2R` of random **unit quaternions**, which yields exact rotations (`det = 1`, `max|R Rᵀ − I| = 0` in float64). Building them with `axis_angle_to_rotation_matrix` instead gives numbers two orders of magnitude worse (`1.455e-06`) — but that is a different defect (the `aa2R` eps, filed separately), not this one.

## Observed

```
R2q(I)          = [1.0000000012499999, 0.0, 0.0, 0.0]
R2q(I, eps=0.0) = [1.0, 0.0, 0.0, 0.0]
||R2q(I)|| - 1  = 1.2499998813808588e-09
worst |q2R(R2q(R)) - R| over 200 exact rotations : 7.327901174747353e-09
worst | ||R2q(R)|| - 1 | over the same 200       : 2.4971434919862645e-09
```

## Expected

`rotation_matrix_to_quaternion` of an exact rotation matrix should return a unit quaternion to working precision: `||q|| − 1` on the order of `1e-16` in float64, not `2.5e-09`. Setting `eps=0.0` already produces exactly that (`[1.0, 0.0, 0.0, 0.0]` for the identity), which shows the algorithm is fine and only the epsilon placement is wrong.

Consequences a user hits:

- `q2R(R2q(R))` is accurate only to `7.3e-09` in float64 for exact rotations, so the round-trip cannot be pinned any tighter than that — a float64 test written at a float64 tolerance fails.
- Downstream code that checks `||q|| == 1` (or normalizes only if it looks non-unit) sees a systematically non-unit quaternion from every call.
- The error floor is dtype-independent in relative terms: `eps = 1e-8` is a fixed absolute quantity, so it dominates float64 precision by seven orders of magnitude while being invisible in float32.

The fix is to move `eps` out of the value and into the guard: clamp the radicand at zero (or at `finfo.tiny`) rather than adding a constant to it, e.g. `sq = torch.sqrt((trace + 1.0).clamp(min=eps)) * 2.0` — which leaves the value exact whenever the radicand is positive and only takes effect where the branch is genuinely degenerate. The `eps` docstring should also be corrected: the divisions are already handled by `safe_zero_division`.

Context: the `+ eps` was introduced to address the NaN gradient reported in #1653 (`sqrt(trace + 1.0)` where the argument can go slightly negative in float32); the proposal there was literally `sq = torch.sqrt(trace + 1.0 + eps) * 2.0`. A `clamp` fixes the same NaN without biasing the value. This is the same defect family as #3939 (`cart2pol`'s `sqrt(x² + y² + eps)`) and the `axis_angle_to_rotation_matrix` eps filed alongside this one — an epsilon folded into a returned value instead of into the gradient path — but a different function with a different magnitude, so it is filed separately.

## Classification

**Broken output, but with a default-changing fix.** "Returns a non-unit quaternion" is wrong under any reading of the function, and the `eps=0.0` path proves the exact answer is one clamp away. However, the fix changes the returned values of **every** call on the default path by `~1e-9`, so it cannot land under byte-identity discipline and it will move any test currently pinned at that precision.

Per the reference-core lifecycle (Part I / Part III §2) my reading is that this is small and unambiguous enough to land as an ordinary broken-output bugfix rather than waiting for the coordinated Repair & Deprecation window — a `1e-9` shift toward the mathematically correct value has no plausible downstream dependent — but since it is technically a default change I am flagging the choice rather than deciding it.

## Environment

kornia 0.9.0rc1 @ `3358ef16`, torch 2.9.1, cpu. Executed from a worktree carrying the batch-3a docs/tests commits on top of `3358ef16`; `rotation_matrix_to_quaternion` is byte-identical to `origin/main` there (no diff hunk lands in it; function body compares AST-identical).

Posted on behalf of @ducha-aiki by Claude (Fable 5).

**Comment by ducha-aiki** [comment]
Scope correction, found while writing the convention pins for this issue (batch 3b of #3941): **the non-unit output is float64-only.** Every lower precision returns exactly `[1, 0, 0, 0]`:

```python
import torch
from kornia.geometry.conversions import rotation_matrix_to_quaternion as R2q

for dtype in (torch.float64, torch.float32, torch.float16, torch.bfloat16):
    q = R2q(torch.eye(3, dtype=dtype))
    print(dtype, q.tolist(), "norm-1 =", float(q.norm() - 1))
```

```
torch.float64  [1.0000000012499999, 0.0, 0.0, 0.0]  norm-1 = 1.2499998813808588e-09
torch.float32  [1.0, 0.0, 0.0, 0.0]                 norm-1 = 0.0
torch.float16  [1.0, 0.0, 0.0, 0.0]                 norm-1 = 0.0
torch.bfloat16 [1.0, 0.0, 0.0, 0.0]                 norm-1 = 0.0
```

The reason is just resolution: `eps = 1e-8` inside the `sqrt` perturbs the result by ~1.25e-9, which is below the spacing of representable numbers near 1.0 at float32 and coarser (`2^-23 ≈ 1.19e-07`), so it rounds away. The mechanism is identical at all dtypes — only float64 is fine-grained enough to show it.

This narrows the practical impact but not the argument for fixing it: float64 is the precision where a downstream `q2R(R2q(R))` round-trip is expected to be exact, and it is the one precision where this is measurably not (worst `|‖q‖−1| = 2.337e-09` over 200 exact rotations, worst round-trip error `7.328e-09`).

The batch-3b PR pins this with a float64-hardcoded strict xfail plus a wart at the current value, so a fix in either direction shows up loudly.

Posted on behalf of @ducha-aiki by Claude (Fable 5).
